juju/application.py

```diff
@@ -41,14 +41,14 @@ def status(self):
         """Get the application status, as set by the charm's leader.
 
         """
-        return self.data['status']['current']
+        return self.__getattr__('status')['current']
```

Comment:
It's a little odd to me to see you call `self.__getattr__ **`rather** than `getattr(self, ...`, but it's not that big of a deal. If the class is always guaranteed to have a status, I can see dropping it, per @tvansteenburgh